src/Smooshable.cpp

```diff
@@ -6,216 +6,245 @@
+}
+
+
+// VDJSmooshable Constructor Functions
 
 /// @brief Creates a Smooshable object for a given V germline gene and read.
 /// @param[in] vgerm_obj
 /// An object of class VGermline.
-/// @param[in] V_left_flexbounds
-/// A 2-tuple of read positions providing the bounds of the V gene's left
-/// flex region.
-/// @param[in] V_right_flexbounds
-/// A 2-tuple of read positions providing the bounds of the V gene's right
-/// flex region.
+/// @param[in] flexbounds
+/// The VDJ flexbounds map from a Query object.
 /// @param[in] emission_indices
 /// A vector of indices corresponding to the observed bases of the read.
-/// @param[in] V_relpos
+/// @param[in] v_relpos
 /// The read position corresponding to the first base of the V germline gene.
 /// @return
-/// A Smooshable object.
-Smooshable VSmooshable(
-    const VGermline& vgerm_obj, std::pair<int, int> V_left_flexbounds,
-    std::pair<int, int> V_right_flexbounds,
-    const Eigen::Ref<const Eigen::VectorXi>& emission_indices, int V_relpos) {
-  // computing the germline match probability matrix and extracting the row
-  // that corresponds to the first match starting position with a germline
-  // state.
-  int start_pos = std::max(V_relpos - V_left_flexbounds.first, 0);
-  Eigen::MatrixXd germ_prob_row =
-      vgerm_obj
-          .GermlineProbMatrix(V_left_flexbounds, V_right_flexbounds,
-                              emission_indices, V_relpos)
-          .row(start_pos);
+/// A SmooshablePtr.
+SmooshablePtr VSmooshable(
+    const VGermline& vgerm_obj,
+    const std::map<std::string, std::pair<int, int>>& flexbounds,
+    const Eigen::Ref<const Eigen::VectorXi>& emission_indices, int v_relpos) {
+  // computing the germline match probability matrix
+  Eigen::MatrixXd germ_prob_matrix = vgerm_obj.GermlineProbMatrix(
+      flexbounds.at("v_l"), flexbounds.at("v_r"), emission_indices, v_relpos);
+
+  // multiplying in the associated landing-out probabilities
+  MultiplyLandingGermProbMatrix(
+      vgerm_obj.landing_out(), germ_prob_matrix, flexbounds.at("v_l"),
+      flexbounds.at("v_r"), v_relpos, vgerm_obj.length(), false);
+
+  // extracting the row that corresponds to the first match starting
+  // position with a germline state
+  int start_pos = std::max(v_relpos - flexbounds.at("v_l").first, 0);
+  germ_prob_matrix = germ_prob_matrix.row(start_pos).eval();
 
   // multiplying in the associated gene and padding probabilities
-  germ_prob_row *= vgerm_obj.gene_prob();
+  germ_prob_matrix *= vgerm_obj.gene_prob();
   double npadding_prob = vgerm_obj.NPaddingProb(
-      V_left_flexbounds, emission_indices, V_relpos, true);
-  germ_prob_row *= npadding_prob;
+      flexbounds.at("v_l"), emission_indices, v_relpos, true);
+  germ_prob_matrix *= npadding_prob;
 
-  // storing Smooshable object
-  Smooshable v_smooshable = Smooshable(germ_prob_row);
-
-  return v_smooshable;
+  /// @todo do we want some move semantics here?
```

Comment:
@matsen @dunleavy005 I agree with "no", it looks like this and `BuildSmooshablePtr` are doing the right thing.